### ISSUE TITLE
Support NOW() function in mongo

### ISSUE THREAD

**Issue by alissapajer** [quasar-analytics/quasar#2502]
Per discussion here https://github.com/quasar-analytics/quasar/issues/2198#issuecomment-316169783 adding this issue to be completed in time for the new mongo release.

**Comment by alissapajer** [comment]
The `NOW()` function exists as a `MapFunc` and so it only needs to be implemented in mongo.

**Comment by rintcius** [comment]
New mongo supports `NOW()` already, but only on mapReduce, not on aggregation pipeline.
I suspect that it's not on the aggregation pipeline because it is represented by [`new Date()` in Bson](https://stackoverflow.com/questions/16455528/mongodb-aggregation-framework-date-now), but the java driver only supports constructing [these with an argument](http://mongodb.github.io/mongo-java-driver/3.4/javadoc/org/bson/BsonDateTime.html), not without an argument.

I _think_ it's possible to add support for this in agg pipeline as well by registering a [codec](http://mongodb.github.io/mongo-java-driver/3.4/bson/codecs/) but before starting implementation I want to double check if we want to go this road.

**Comment by rintcius** [comment]
Added IT [here](https://github.com/rintcius/quasar/tree/2502-mongo-now) on top of @sellout's new mongo branch

**Comment by jdegoes** [comment]
In theory, in batch mode, `NOW()` can be replaced with the current date / time before the query is handed off to the backend.

**Comment by rintcius** [comment]
@jdegoes Yes, indeed I was thinking about that as well. I was thinking that it may even be better, e.g. in case we are doing cross-server queries (which lead me to think that it would be nice to have query hints for these kind of things). 

BTW there are currently different aliases for `now`. We also choose one of them to have that meaning instead:

```
  CIName("clock_timestamp")         -> date.Now, // Postgres (instantaneous)
  CIName("current_timestamp")       -> date.Now, // *, SQL92
  CIName("getdate")                 -> date.Now, // SQL Server
  CIName("localtimestamp")          -> date.Now, // MySQL, Postgres (trans start)
  CIName("now")                     -> date.Now, // MySQL, Postgres (trans start)
  CIName("statement_timestamp")     -> date.Now, // Postgres (statement start)
  CIName("transaction_timestamp")   -> date.Now, // Postgres (trans start)
```

**Comment by zeeiyerWork** [comment]
@rintcius - reassigned this to @jsantos17 -> since he has a little bit more time in August.

**Comment by rintcius** [comment]
@zeeiyerWork This one I already discussed with @wemrysi & worked on a bit - would be more efficient if I could do it. 

@jsantos17 if you come to this first then let's chat before you start on it

**Comment by alissapajer** [comment]
@jsantos17 now that https://github.com/quasar-analytics/quasar/pull/2704 has merged, can we update the status to "fixed in new mongo"?

**Comment by jsantos17** [comment]
Confirmed as working
